blurring_as_a_service/inference_pipeline/components/blur_images.py

```diff
@@ -37,26 +39,63 @@ def blur_images(
 
     Parameters
     ----------
-    data_to_blur:
-        Images to be blurred
+    folder:
+        Path of the mounted folder containing the images.
+    files_to_blur:
+        Text file containing multiple rows where each row has a relative path,
+        taking folder as root and the path to the image.
     results_detection:
-        Areas to blur determined by the model.
+        Path of the mounted folder containing the result of the detection areas determined by the model.
     results_path:
-        Where to store the results.
+        Path of the mounted folder where to store the results.
     """
-    detection_result_files = os.listdir(f"{results_detection}/detection_result/labels")
     detection_result_files = [
-        os.path.splitext(os.path.basename(path))[0] for path in detection_result_files
+        os.path.splitext(os.path.basename(path))[0]
+        for path in os.listdir(f"{results_detection}/detection_result/labels")
     ]
-    for image_to_blur in os.listdir(data_to_blur):
-        image_to_blur_filename = os.path.splitext(image_to_blur)[0]
-        if image_to_blur_filename in detection_result_files:
-            ImageBlurrer(f"{data_to_blur}/{image_to_blur}").blur_and_store(
-                f"{results_detection}/detection_result/labels/{image_to_blur_filename}.txt",
-                f"{results_path}/detection_result/{os.path.basename(image_to_blur)}",
-            )
-        else:
-            shutil.copyfile(
-                f"{data_to_blur}/{image_to_blur}",
-                f"{results_path}/detection_result/{os.path.basename(image_to_blur)}",
+
+    with open(files_to_blur, "r") as images_path_to_blur:
+        for image_to_blur in images_path_to_blur:
+            blur_image(
+                image_to_blur,
+                detection_result_files,
+                folder,
+                results_detection,
+                results_path,
             )
+
+
+def blur_image(
+    image_to_blur: str,
+    detection_result_files: List[str],
+    folder: str,
+    results_detection: str,
+    results_path: str,
```

Comment:
I think the arguments of this function are a bit confusing. Is it possible to infer some of them? 
At least the `folder` one could be renamed. 
And doesn't `results_detection` contain `detection_result_files`?